core/src/test/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilderTest.java

```diff
@@ -0,0 +1,346 @@
+        GeoBoundingBoxQueryBuilder builder = new GeoBoundingBoxQueryBuilder("mapped_geo");
+        Rectangle box = RandomShapeGenerator.xRandomRectangle(getRandom(), RandomShapeGenerator.xRandomPoint(getRandom()));
+
+        if (frequently()) {
+            // check the top-left/bottom-right combination of setters
+            int path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.topLeft(new GeoPoint(box.getMaxY(), box.getMinX()));
+                break;
+            case 1:
+                builder.topLeft(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMinX()));
+                break;
+            default:
+                builder.topLeft(box.getMaxY(), box.getMinX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.bottomRight(new GeoPoint(box.getMinY(), box.getMaxX()));
+                break;
+            case 1:
+                builder.bottomRight(GeohashUtils.encodeLatLon(box.getMinY(), box.getMaxX()));
+                break;
+            default:
+                builder.bottomRight(box.getMinY(), box.getMaxX());
+            }
+        } else {
+            // check the deprecated top-right/bottom-left combination of setters
+            int path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.topRight(new GeoPoint(box.getMaxY(), box.getMaxX()));
+                break;
+            case 1:
+                builder.topRight(GeohashUtils.encodeLatLon(box.getMaxY(), box.getMaxX()));
+                break;
+            default:
+                builder.topRight(box.getMaxY(), box.getMaxX());
+            }
+
+            path = randomIntBetween(0, 2);
+            switch (path) {
+            case 0:
+                builder.bottomLeft(new GeoPoint(box.getMinY(), box.getMinX()));
+                break;
+            case 1:
+                builder.bottomLeft(GeohashUtils.encodeLatLon(box.getMinY(), box.getMinX()));
+                break;
+            default:
+                builder.bottomLeft(box.getMinY(), box.getMinX());
+            }
+        }
+
+        if (randomBoolean()) {
+            builder.coerce(randomBoolean());
+        }
+        if (randomBoolean()) {
+            builder.ignoreMalformed(randomBoolean());
+        }
+
+        builder.type(randomFrom(Type.values()));
+        return builder;
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testValidation_empty_fieldname() {
+        new GeoBoundingBoxQueryBuilder("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testValidation_null_fieldname() {
+        new GeoBoundingBoxQueryBuilder(null);
+    }
+
+    @Test
+    @Override
+    public void testToQuery() throws IOException {
+        QueryShardContext context = createShardContext();
```

Comment:
Thanks for pointing me to assumeTrue - hadn't known about it before.

> And test the exception thrown when there's no types separately  (if we don't do that already)?

First of all I don't see a test where we do that already. While trying to create a test for this specifically I ran into a question:

Solution a) Easiest solution would be to use the above mentioned assumeTrue, just with the assumption reversed. But that would be executed only once in a while.

Solution b) I tried to create a test case that deterministically triggers the exception. If I'm reading things correctly the problem is that when the test method is executed, currentTypes is already bound into the mapperService, referenced by queryParseService, which ultimately gets linked into the QueryShardContext. If queryParseService wouldn't come out of some Guice injection I'd go for simply supplying a separate one just for the exception testing - however it's being created through Guice.

So after some digging I'm unsure how to proceed: Did I miss something obvious on how to create such a test deterministically? Should I go for option a and stop spending more time on digging here? Should I continue digging and give myself a bigger timebox to see if option b can be achieved?